code/modules/reagents/reagents/toxin.dm

```diff
@@ -555,18 +564,48 @@
 	RegisterSignal(L, COMSIG_HUMAN_DAMAGE_TAKEN, .proc/transvitox_human_damage_taken)
 
 /datum/reagent/toxin/xeno_transvitox/on_mob_life(mob/living/L, metabolism)
-	if(prob(25))
-		to_chat(L, "<span class='warning'>You notice being strangely revitalised.</span>")
+	if(prob(10))
+		to_chat(L, "<span class='warning'>You notice your wounds crusting over with disgusting green ichor.</span>")
+
+	var/fire_loss = L.getFireLoss()
+	if(!fire_loss) //If we have no burn damage, cancel out
+		return ..()
+
+	var/tox_cap_multiplier = 1
+
+	if(L.reagents.get_reagent_amount(/datum/reagent/toxin/xeno_hemodile)) //Each other Defiler toxin doubles the multiplier
+		tox_cap_multiplier *= 2
+
+	if(L.reagents.get_reagent_amount(/datum/reagent/toxin/xeno_neurotoxin))
+		tox_cap_multiplier *= 2
+
+	var/tox_loss = L.getToxLoss()
+	if(tox_loss > DEFILER_TRANSVITOX_CAP) //If toxin levels are already at their cap, cancel out
+		return ..()
+
+	var/dam = (current_cycle * 0.25 * tox_cap_multiplier) //Converts burn damage at this rate to toxin damage
+
+	if(fire_loss < dam) //If burn damage is less than damage to be converted, have the conversion value be equal to the burn damage
+		dam = fire_loss
+
+	L.heal_limb_damage(burn = dam, updating_health = TRUE) //Heal damage equal to toxin damage dealt; heal before applying toxin damage so we don't flash kill the target
+	L.adjustToxLoss(dam)
+
 	return ..()
 
 /datum/reagent/toxin/xeno_transvitox/proc/transvitox_human_damage_taken(mob/living/L, damage)
 	SIGNAL_HANDLER
-	var/dam = min(damage*0.4, 45 - L.getToxLoss()) // hard caps damage conversion to not exceed 45 tox
-	if((L.getBruteLoss() + L.getFireLoss()) < dam)
-		return
-	L.adjustToxLoss(dam)
-	var/healed_brute = min(dam, L.getBruteLoss())
-	L.heal_limb_damage(healed_brute)
-	if(!L.getFireLoss())
+
+	var/tox_cap_multiplier = 1
+
+	if(L.reagents.get_reagent_amount(/datum/reagent/toxin/xeno_hemodile)) //Each other Defiler toxin doubles the multiplier
+		tox_cap_multiplier *= 2
+
+	if(L.reagents.get_reagent_amount(/datum/reagent/toxin/xeno_neurotoxin))
+		tox_cap_multiplier *= 2
+
+	var/tox_loss = L.getToxLoss()
+	if(tox_loss > DEFILER_TRANSVITOX_CAP) //If toxin levels are already at their cap, cancel out
 		return
-	L.heal_limb_damage(0, dam - healed_brute)
+
+	L.setToxLoss(max(tox_loss + min(L.getBruteLoss() * 0.15 * tox_cap_multiplier, damage * 0.15 * tox_cap_multiplier),tox_loss))  //Deal bonus tox damage equal to a % of the lesser of the damage taken or the target's brute damage; capped at 180.
```

Comment:
This is very very strong, thats 15% extra damage of every brute damage taken doenst matter the source, and can be doubled twice to 60% increasing 60% damage of all sources,   this should be way way lower, maybe 5% of brute damage taken up to 15% ish

Reply:
This is what the TM is for

Reply:
still standing